Autorest Regen Preview to 3.0.0-alpha.20230713.1 by Mingzhe Huang from refs/pull/3584/head

azure-sdk opened the pull request:
Triggered from https://github.com/Azure/autorest.csharp/pull/3584

azure-sdk committed:
Update Generator Version 3.0.0-alpha.20230713.1

azure-sdk committed:
Update SDK codes

azure-sdk committed:
Update SDK codes

azure-sdk committed:
Update SDK codes

azure-sdk committed:
Update SDK codes

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Health.Insights.ClinicalMatching](https://apiview.dev/Assemblies/Review/698325eac58749498fe05898ccb82685)

archerzz committed:
fix `Azure.Analytics.Purview.Share` issues

- update api signature
- add repeatable request headers into ignored headers of tests

archerzz commented:
> **API change check**
> 
> APIView has identified API level changes in this PR and created following API reviews.
> 
> [Azure.Health.Insights.ClinicalMatching](https://apiview.dev/Assemblies/Review/698325eac58749498fe05898ccb82685) [Azure.Health.Insights.CancerProfiling](https://apiview.dev/Assemblies/Review/ceeae1e020a54a66a793f8aa88215271) [Azure.Analytics.Purview.Sharing](https://apiview.dev/Assemblies/Review/b4256e95df1f427bb7bfd4e754e2aa44) [Azure.Analytics.Purview.Share](https://apiview.dev/Assemblies/Review/e03504e0eb054127ba825664e7e9ee3e)

For those SDKs:
- `Azure.Health.Insights.ClinicalMatching`: still in beta
- `Azure.Health.Insights.ClinicalMatching`: still in beta
- `Azure.Analytics.Purview.Share`: still in beta
- `Azure.Analytics.Purview.Share`: still in beta and deprecated, see [here](https://www.nuget.org/packages/Azure.Analytics.Purview.Share/1.0.3-beta.20)

So we can safely update their API signatures.

archerzz committed:
update api signature of `Azure.Analytics.Purview.Sharing`

archerzz committed:
fix healthinsight api signature

archerzz committed:
fix `Azure.Communication.Chat` issues

archerzz committed:
fix `Azure.Commuinication.Rooms`

archerzz committed:
fix wrong `Repeatability-First-Sent` format in `Azure.Communication.CallingServer`

archerzz committed:
fix `Azure.Communication.CallAutomation`

- fix API invocation due to removal of repeatability header parameters
- remove `RepeatabilityHeaders` class and its tests since it is not necessary

lirenhe commented on sdk/healthinsights/Azure.Health.Insights.CancerProfiling/api/Azure.Health.Insights.CancerProfiling.netstandard2.0.cs:
```diff
@@ -6,10 +6,10 @@ public partial class CancerProfilingClient
-        public virtual Azure.Operation<Azure.Health.Insights.CancerProfiling.OncoPhenotypeResult> InferCancerProfile(Azure.WaitUntil waitUntil, Azure.Health.Insights.CancerProfiling.OncoPhenotypeData oncoPhenotypeData, string repeatabilityRequestId = null, System.DateTimeOffset? repeatabilityFirstSent = default(System.DateTimeOffset?), System.Threading.CancellationToken cancellationToken = defaul ...
-        public virtual System.Threading.Tasks.Task<Azure.Operation<System.BinaryData>> InferCancerProfileAsync(Azure.WaitUntil waitUntil, Azure.Core.RequestContent content, string repeatabilityRequestId = null, System.DateTimeOffset? repeatabilityFirstSent = default(System.DateTimeOffset?), Azure.RequestContext context = null) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Operation<Azure.Health.Insights.CancerProfiling.OncoPhenotypeResult>> InferCancerProfileAsync(Azure.WaitUntil waitUntil, Azure.Health.Insights.CancerProfiling.OncoPhenotypeData oncoPhenotypeData, string repeatabilityRequestId = null, System.DateTimeOffset? repeatabilityFirstSent = default(System.DateTimeOffset?), System.Threading.Cancellati ...
+        public virtual Azure.Operation<System.BinaryData> InferCancerProfile(Azure.WaitUntil waitUntil, Azure.Core.RequestContent content, Azure.RequestContext context = null) { throw null; }
```
I think this is the right fix but this would cause breaking change to health insight.  @m-nash, do you have any concern for this change?

archerzz replied:
> I think this is the right fix but this would cause breaking change to health insight. @m-nash, do you have any concern for this change?

They are all beta versions. I think it's OK to change it.

m-nash replied:
yeah since they are in beta this is fine and expected

asaflevi-ms replied:
Hi,
It is ok by us that the repeatability headers will be set internally.
Related to the above change, I see that the Sync method was generated but the Async method dropped.
Any reason why?

Also, a question: is there an option to set the repeatability headers as context policy? (not see it used, it is just my curiosity)

archerzz replied:
Hi @asaflevi-ms are you referring to [ClinicalMatchingClient](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/healthinsights/Azure.Health.Insights.ClinicalMatching/src/Generated/ClinicalMatchingClient.cs)? Which method do you refer to?
There are 4 methods being changed. 2 of them are sync methods, and other 2 are async methods. 

As for the policy, the answer is yes. Users are free to set a pipeline policy through `RequestContext`. See here for example: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/RequestContext.md#add-policy-for-request

asaflevi-ms replied:
Hi @archerzz 
My mistake, I missed it.
I reviewed HealthInsights clients (ClinicalMatchingClient and CancerProfilingClient).
Approved by me.

archerzz committed:
temporarily exclude repeatability request headers from purview testing

archerzz committed:
re-recording for `Azure.Health.Insights.CancerProfiling`

archerzz committed:
re-recording for `Azure.Health.Insights.ClinicalMatching`

archerzz committed:
temporarily exclude repeatability request headers from `Azure.Communication.CallingServer` tests

archerzz committed:
Merge branch 'main' into auto-update-autorest-3.0.0-alpha.20230713.1

archerzz committed:
add backward compatbility to `Azure.Communication.Chat`

- reverts commit 015e75d543ea9fd8b2455d8076cc9f66c7129a5e
- add backward compatible methods of `ChatRestClient` which still accepts
incoming `repeatability-request-id`
- revert change on test case base

archerzz committed:
Merge branch 'main' into auto-update-autorest-3.0.0-alpha.20230713.1

archerzz committed:
modify `Azure.Communication.CallingServer` test recording

- update test recordings to add repeatability request headers
- change test case base to use `IgnoredHeaders` instead of `LegacyExcludedHeaders`

archerzz committed:
modify `Azure.Analytics.Purview.Share` test recording

- update test recordings to add repeatability request headers
- change test case base to use `IgnoredHeaders` instead of `LegacyExcludedHeaders`

archerzz committed:
Merge branch 'main' into auto-update-autorest-3.0.0-alpha.20230713.1

archerzz commented:
Close since the changes have been merged in https://github.com/Azure/azure-sdk-for-net/pull/37746
